front/lib/api/assistant/get_favorite_states.ts

```diff
@@ -0,0 +1,33 @@
+import { Op } from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import { AgentUserRelation } from "@app/lib/models/assistant/agent";
+
+export async function getFavoriteStates({
+  auth,
+  configurationSIds,
+}: {
+  auth: Authenticator;
+  configurationSIds: string[];
+}): Promise<Map<string, boolean>> {
+  const [userId, workspaceId] = [auth.user()?.id, auth.workspace()?.id];
```

Comment:
You will never have a userId through the public API.

Also you can do:
```
auth.getNonNullableWorkspace()
```

And same for the user.